src/SignalR/server/Core/src/HubInvocationContext.cs

```diff
@@ -30,31 +40,51 @@ public HubInvocationContext(HubCallerContext context, Type hubType, string hubMe
         /// <param name="context">Context for the active Hub connection and caller.</param>
         /// <param name="hubMethodName">The name of the Hub method being invoked.</param>
         /// <param name="hubMethodArguments">The arguments provided by the client.</param>
+        [Obsolete("This constructor is obsolete and will be removed in a future version. The recommended alternative is to use the other constructor.")]
         public HubInvocationContext(HubCallerContext context, string hubMethodName, object[] hubMethodArguments)
         {
+#pragma warning disable CS0618 // Type or member is obsolete
             HubMethodName = hubMethodName;
-            HubMethodArguments = hubMethodArguments;
+#pragma warning restore CS0618 // Type or member is obsolete
+            Arguments = hubMethodArguments;
             Context = context;
         }
 
+        internal HubInvocationContext(ObjectMethodExecutor objectMethodExecutor, HubCallerContext context, IServiceProvider serviceProvider, Hub hub, object[] hubMethodArguments)
+            : this(context, serviceProvider, hub, objectMethodExecutor.MethodInfo, hubMethodArguments)
+        {
+            ObjectMethodExecutor = objectMethodExecutor;
+        }
+
         /// <summary>
         /// Gets the context for the active Hub connection and caller.
         /// </summary>
         public HubCallerContext Context { get; }
 
         /// <summary>
-        /// Gets the Hub type.
+        /// Gets the Hub instance.
         /// </summary>
-        public Type HubType { get; }
+        public Hub Hub { get; }
 
         /// <summary>
         /// Gets the name of the Hub method being invoked.
         /// </summary>
+        [Obsolete("This property is obsolete and will be removed in a future version. The recommended alternative is to use HubMethod.Name.")]
         public string HubMethodName { get; }
 
         /// <summary>
         /// Gets the arguments provided by the client.
         /// </summary>
-        public IReadOnlyList<object> HubMethodArguments { get; }
+        public IReadOnlyList<object> HubMethodArguments { get => Arguments; }
```

Comment:
```suggestion
        public IReadOnlyList<object> HubMethodArgument => Arguments;
```

Reply:
What about just `Arguments` for the public property?

Reply:
I'd rather not make another breaking change